provider/pkg/provider/provider_2e2_test.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.
+
+package provider_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/pulumi/providertest"
+	"github.com/pulumi/providertest/pulumitest"
+	"github.com/pulumi/providertest/pulumitest/assertpreview"
+	"github.com/pulumi/providertest/pulumitest/assertrefresh"
+	"github.com/pulumi/providertest/pulumitest/opttest"
+	"github.com/pulumi/pulumi-aws-native/provider/pkg/provider"
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+)
+
+func TestE2eSnapshots(t *testing.T) {
+	t.Run("logGroup-0.94.0", func(t *testing.T) {
+		t.Parallel()
+		test := newAwsTest(t, filepath.Join("testdata", "logGroup"))
+		testUpgradeFrom(test, "0.94.0")
+	})
+
+	t.Run("webAcl-0.94.0", func(t *testing.T) {
+		t.Parallel()
+		test := newAwsTest(t, filepath.Join("testdata", "webAcl"))
+		testUpgradeFrom(test, "0.94.0")
+	})
+}
+
+func testUpgradeFrom(test *pulumitest.PulumiTest, version string) {
+	test.T().Helper()
+	result := providertest.PreviewProviderUpgrade(test, "aws-native", version)
+	assertpreview.HasNoChanges(test.T(), result)
+}
+
+func testLifecycle(test *pulumitest.PulumiTest) auto.OutputMap {
```

Comment:
this method is unused for now?

Reply:
Yes, until we re-instate the new style integration tests. Will tidy